prover/crates/bin/prover_autoscaler/src/global/scaler.rs

```diff
@@ -160,19 +155,55 @@ impl<K: Key> Scaler<K> {
             .flat_map(|c| self.convert_to_pool(namespace, c))
             .collect();
 
-        pools.sort_by(|a, b| {
-            a.key
-                .cmp(&b.key) // Sort by Key first.
-                .then(
-                    a.sum_by_pod_status(PodStatus::NeedToMove)
-                        .cmp(&b.sum_by_pod_status(PodStatus::NeedToMove)),
-                ) // Sort by need to evict.
-                .then(
-                    a.sum_by_pod_status(PodStatus::LongPending)
-                        .cmp(&b.sum_by_pod_status(PodStatus::LongPending)),
-                ) // Sort by long Pending pods.
-                .then(a.scale_errors.cmp(&b.scale_errors)) // Sort by scale_errors in the cluster.
-                .then(
+        // If a pool has NeedToMove pod, max_pool_size is set to number of Running+Pending pods.
+        for pool in &mut pools {
+            if pool.sum_by_pod_status(PodStatus::NeedToMove) > 0 {
+                pool.max_pool_size = pool.sum_by_pod_status(PodStatus::Running)
+                    + pool.sum_by_pod_status(PodStatus::Pending);
+                tracing::debug!(
+                    "Pool {}:{:?} has NeedToMove pods, max_pool_size adjusted to {}",
+                    pool.name,
+                    pool.key,
+                    pool.max_pool_size
+                );
+            }
+        }
+
+        let priority_cmp = |a: &Pool<K>, b: &Pool<K>| {
```

Comment:
why don't we implement Ord explicitly?

Reply:
`priority_cmp` is just a part of ordering. I'll try to implement complete Ord.

Reply:
Actually it becomes way too complicated: to sort pools it needs `cluster_priority` and `target_priority`.